portfolio/src/main/java/com/google/sps/servlets/ReadGameServlet.java

```diff
@@ -32,7 +32,15 @@
 @WebServlet("/view-games")
 public class ReadGameServlet extends HttpServlet {
 
-    Game game = new Game();
+    String searchTerm;
+
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        //gets input from form
+        searchTerm = request.getParameter("search");
+
```

Comment:
Rather than redirecting to view-games, you can just search for the game and return the game found right here then in the UI change that part of the display based on the response. I think that might be cleaner. I don't think it was discussed in the training, but it is also good to think about what would happen if multiple people are using your page at the same time. If you do the search and return it here, then multiple people can see different games without the server overwriting its data.